src/es-module-shims.js

```diff
@@ -54,15 +54,34 @@ async function topLevelLoad (url, source, polyfill) {
   lastLoad = undefined;
   resolveDeps(load, seen);
   // inline "module-shim" must still execute even if no shim
-  if (source && !polyfill && !load.n)
-    return dynamicImport(createBlob(source));
+  if (source && !polyfill && !load.n) {
+    const module = dynamicImport(createBlob(source));
+    revokeObjectURLs(Object.keys(seen));
+    return module;
+  }
   const module = await dynamicImport(load.b);
   // if the top-level load is a shell, run its update function
-  if (load.s)
+  if (load.s) {
     (await dynamicImport(load.s)).u$_(module);
+  }
+  revokeObjectURLs(Object.keys(seen));
   return module;
 }
 
+function revokeObjectURLs(registryKeys) {
```

Comment:
Wondering if we should make this async and have an async tick eg every 100 revocations?